raiden/connection_manager.py

```diff
@@ -265,39 +269,84 @@ def _open_channels(self):
         Note:
             - This method must be called with the lock held.
         """
-        open_channels = views.get_channelstate_open(
-            chain_state=views.state_from_raiden(self.raiden),
-            payment_network_id=self.registry_address,
-            token_address=self.token_address,
-        )
 
-        qty_channels_to_open = self.initial_channel_target - len(open_channels)
-        if qty_channels_to_open <= 0:
-            return
-
-        for partner in self.find_new_partners(qty_channels_to_open):
-            try:
-                self.api.channel_open(
-                    self.registry_address,
-                    self.token_address,
-                    partner,
-                )
-            except DuplicatedChannelError:
-                # This can fail because of a race condition, where the channel
-                # partner opens first.
-                log.info('partner opened channel first')
-
-            try:
-                self.api.set_total_channel_deposit(
-                    self.registry_address,
-                    self.token_address,
-                    partner,
-                    self._initial_funding_per_partner,
+        while True:
```

Comment:
just a nitpick: could this be split into smaller functions?

Reply:
Ok, split `_join_partner` to its own method, but as the other parts of this function are kind of chained together, I couldn't get an easy way to split them. I think it's small enough now.